File: src/2022/06/b.ts
import * as fs from "fs";

fs.readFile(process.argv[2] ?? 'in.txt', 'utf-8', (err, data) => {
    if (err) console.log(err);

    const splitData = data.replace("\n", "").split("");
    for (let i = 0; i < splitData.length-3; i++) {
        if(new Set([...splitData.slice(i, i+14)]).size === 14) {
            console.log(i+14);
            break;
        }
    }
});

File: src/2022/06/b.go
package main

import (
	"os"
	"strings"
)

func main() {
	file, _ := os.ReadFile("in.txt")
	splitInput := strings.Split(string(file), "")
	for i := range splitInput {
		set := map[string]bool{}
		for j := i; j < i+14; j++ {
			set[splitInput[j]] = true
		}
		if len(set) == 14 {
			println(i+14)
			break
		}
	}

}
